test/project_types/extension/features/argo/base_test.rb

```diff
@@ -0,0 +1,223 @@
+            error = assert_raises(ShopifyCli::Abort) { @dummy_argo.config(@context) }
+            assert error
+              .message
+              .include?(
+                @context.message(
+                  'features.argo.dependencies.argo_missing_renderer_package_error',
+                  error_message
+                )
+              )
+          end
+        end
+
+        def test_runs_yarn_install_and_yarn_run_script_if_the_package_manager_is_yarn
+          with_stubbed_script(@context, Features::Argo::Base::SCRIPT_PATH) do
+            ShopifyCli::JsSystem.any_instance.stubs(:package_manager).returns('yarn')
+            Argo::Base.any_instance.expects(:run_yarn_install).returns(true).once
+            Argo::Base.any_instance.expects(:run_yarn_run_script).returns(true).once
+            Argo::Base.any_instance.stubs(:extract_argo_renderer_version).returns("0.0.1")
+
+            @dummy_argo.config(@context)
+          end
+        end
+
+        def test_does_not_run_yarn_install_and_yarn_run_script_if_the_package_manager_is_npm
+          with_stubbed_script(@context, Features::Argo::Base::SCRIPT_PATH) do
+            ShopifyCli::JsSystem.any_instance.stubs(:package_manager).returns('npm')
+            Argo::Base.any_instance.expects(:run_yarn_install).never
+            Argo::Base.any_instance.expects(:run_yarn_run_script).never
+            Argo::Base.any_instance.stubs(:extract_argo_renderer_version).returns("0.0.1")
+
+            @dummy_argo.config(@context)
+          end
+        end
+
+        def test_aborts_with_error_if_yarn_install_command_fails
+          with_stubbed_script(@context, Features::Argo::Base::SCRIPT_PATH) do
+            ShopifyCli::JsSystem.any_instance.stubs(:package_manager).returns('yarn')
+            ShopifyCli::JsSystem.any_instance.stubs(:call).returns([@result, @error_message, mock(success?: false)])
+            error = assert_raises(ShopifyCli::Abort) { @dummy_argo.config(@context) }
+            assert error.message.include?(
+              @context.message('features.argo.dependencies.yarn_install_error', @error_message)
+            )
+          end
+        end
+
+        def test_aborts_with_error_if_yarn_run_script_command_fails
+          with_stubbed_script(@context, Features::Argo::Base::SCRIPT_PATH) do
+            ShopifyCli::JsSystem.any_instance.stubs(:package_manager).returns('yarn')
+            Argo::Base.any_instance.stubs(:run_yarn_install).returns(true)
+            ShopifyCli::JsSystem.any_instance.stubs(:call).returns([@result, @error_message, mock(success?: false)])
+            error = assert_raises(ShopifyCli::Abort) { @dummy_argo.config(@context) }
+            assert error.message.include?(
+              @context.message('features.argo.dependencies.yarn_run_script_error', @error_message)
+            )
+          end
+        end
+
+        def test_raises_not_implemented_error_for_setup_method
```

Comment:
I don't think this test applies anymore. We should probably add a similar one for the `git_template` method.